projects/vdk-core/tests/functional/ingestion/test_ingest_chain.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Functional test aiming at verifying the end-to-end operation of the
+ingestion plugins chaining functionality
+"""
+import logging
+from typing import List
+
+from click.testing import Result
+from functional.ingestion import utils
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_input import IIngesterPlugin
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.plugin.test_utils.util_funcs import cli_assert_equal
+from vdk.plugin.test_utils.util_funcs import CliEntryBasedTestRunner
+from vdk.plugin.test_utils.util_plugins import IngestIntoMemoryPlugin
+
+
+log = logging.getLogger(__name__)
+
+
+class ConvertPayloadValuesToString(IIngesterPlugin):
+    def pre_ingest_process(self, payload: List[dict]) -> List[dict]:
+        return [{k: str(v) for (k, v) in i.items()} for i in payload]
+
+    @hookimpl
+    def initialize_job(self, context: JobContext) -> None:
+        log.info("Initialize data job with ConvertPayloadValuesToString Plugin.")
+
+        context.ingester.add_ingester_factory_method("convert-to-string", lambda: self)
+
+
+def test_chained_ingest():
```

Comment:
Shouldn't the method has configuration ? 
```
INGEST_PLUGIN_PROCESS_SEQUENCE=convert-to-string,memory
```

another alternative
```
INGEST_PLUGIN_PROCESS_SEQUENCE=convert-to-string
```

and so on for the other cases ....

Reply:
Good catch. Fixed, thanks

Reply:
I think we'd need test for the other cases. 
cases: 
 
1. INGEST_PLUGIN_PROCESS_SEQUENCE=A,B in job code no method set
2. INGEST_PLUGIN_PROCESS_SEQUENCE=A,B in job code method = C
3. INGEST_PLUGIN_PROCESS_SEQUENCE=A,B and VDK_INGEST_METHOD_DEFAULT=C
4. INGEST_PLUGIN_PROCESS_SEQUENCE=A,B and  VDK_INGEST_METHOD_DEFAULT=C and job code is D
5. INGEST_PLUGIN_PROCESS_SEQUENCE=A,B and in job code method = B also (which I think is the current shown case)
6. INGEST_PLUGIN_PROCESS_SEQUENCE="" 

But as long as we are on the same page on their behaviour we can do the actual tests later: 
So what is expected behavior?
Maybe?
1. A,B are used for pre-processing, B is used for ingesting and post-procesing 
2. A,B is ...